File: 01-arrays-and-hashing/top-k-frequent-elements/py/main.py
class Solution:
    def topKFrequent(self, nums: List[int], k: int) -> List[int]:
        hmap = {}

        for n in nums:
            if n in hmap:
                hmap[n].append(n)
            else:
                hmap[n] = [n]

        res = list(hmap.values())
        res.sort(key=len, reverse=True)

        v = []

        for l in res:
            v.append(l[0])

        return v[:k]

File: 01-arrays-and-hashing/top-k-frequent-elements/go/main.go
package main

import (
	"sort"
)

func main() {
}

func topKFrequent(nums []int, k int) []int {
	frequentTracker := make(map[int]int)

	for _, num := range nums {
		frequentTracker[num]++
	}

	countedNums := make([]int, 0, len(frequentTracker))

	for key := range frequentTracker {
		countedNums = append(countedNums, key)
	}

	sort.Slice(countedNums, func(i, j int) bool {
		return frequentTracker[countedNums[i]] > frequentTracker[countedNums[j]]
	})

	l := min(k, len(countedNums))

	numsSlice := countedNums[:l]

	return numsSlice
}
